airbyte-ci/connectors/pipelines/pipelines/builds/java_connectors.py

```diff
@@ -11,36 +11,42 @@
 
 
 class BuildConnectorDistributionTar(GradleTask):
+    """
+    A step to build a Java connector image using the distTar Gradle task.
+    """
+
     title = "Build connector tar"
     gradle_task_name = "distTar"
 
+
     async def _run(self) -> StepResult:
-        cdk_includes = ["./airbyte-cdk/java/airbyte-cdk/**"]
-        with_built_tar = (
-            environments.with_gradle(
-                self.context,
-                self.build_include + cdk_includes,
-            )
-            .with_exec(["./gradlew", ":airbyte-cdk:java:airbyte-cdk:publishSnapshotIfNeeded"])
-            .with_mounted_directory(str(self.context.connector.code_directory), await self.context.get_connector_dir())
-            .with_exec(self._get_gradle_command())
-            .with_workdir(f"{self.context.connector.code_directory}/build/distributions")
-        )
+        result = await super()._run()
+        if result.status is not StepStatus.SUCCESS:
+            return result
+
+        with_built_tar = result.output_artifact.with_workdir(f"{self.context.connector.code_directory}/build/distributions")
         distributions = await with_built_tar.directory(".").entries()
         tar_files = [f for f in distributions if f.endswith(".tar")]
-        await self._export_gradle_dependency_cache(with_built_tar)
         if len(tar_files) == 1:
             return StepResult(
                 self,
                 StepStatus.SUCCESS,
-                stdout="The tar file for the current connector was successfully built.",
+                stderr="The distribution tar file for the current java connector was built.",
```

Comment:
```suggestion
                stdout="The distribution tar file for the current java connector was built.",
```
If the step is successful I'd expect the success message to be stored in the `stdout` attribute. If think we can safely return the `result` variable following this check instead of creating an artificial StepResult. And feel free to remove the `len(tar_files)` check if its something that can never happen in the java land :D.

Reply:
This change was a mistake. Fixing. I think it's OK to be a little defensive here about the number of tar files. You're right that this is unlikely to happen, though.